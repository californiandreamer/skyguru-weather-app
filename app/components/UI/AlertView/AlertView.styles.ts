import { transparentGray } from 'app/constants/colors'
import { gap, largeGap } from 'app/constants/values'
import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  container: {
    padding: largeGap,
  },
  connectionIcon: {
    padding: gap,
    justifyContent: 'center',
    alignItems: 'center',
  },
  textArea: {
    paddingVertical: gap,
  },
  button: {
    padding: gap,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: transparentGray,
  },
})
