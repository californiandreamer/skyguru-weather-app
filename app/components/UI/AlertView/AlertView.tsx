import StyledText from 'app/components/HOC/Text'
import {
  largeIconSize,
  pressOpacity,
  smallFontSize,
} from 'app/constants/values'
import { RootState } from 'app/store/reducers/rootReducer'
import React from 'react'
import { TouchableOpacity, View } from 'react-native'
import { useSelector } from 'react-redux'

import styles from './AlertView.styles'

const AlertView: React.FC = () => {
  const {
    illustration: Illustration,
    text,
    action,
  } = useSelector((state: RootState) => state.alert.props)

  const renderIcon = () => (
    <View style={styles.connectionIcon}>
      {Illustration ? (
        <Illustration width={largeIconSize} height={largeIconSize} />
      ) : null}
    </View>
  )

  const renderText = () => (
    <View style={styles.textArea}>
      <StyledText fontSize={smallFontSize}>{text}</StyledText>
    </View>
  )

  const renderButton = () => (
    <TouchableOpacity
      style={styles.button}
      activeOpacity={pressOpacity}
      onPress={() => (action ? action() : null)}
    >
      <StyledText>Try again</StyledText>
    </TouchableOpacity>
  )

  return (
    <View style={styles.container}>
      {renderIcon()}
      {renderText()}
      {action ? renderButton() : null}
    </View>
  )
}

export default AlertView
