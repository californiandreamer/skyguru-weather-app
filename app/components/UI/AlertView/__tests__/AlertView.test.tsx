import configureStore from 'app/store'
import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

import AlertView from '../AlertView'

describe('AlertView component', () => {
  it('renders correctly', () => {
    const store = configureStore()

    const tree = renderer
      .create(
        <Provider store={store}>
          <AlertView />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
