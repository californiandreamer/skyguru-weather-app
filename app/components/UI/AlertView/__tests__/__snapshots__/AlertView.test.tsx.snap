// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertView component renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 40,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  />
  <View
    style={
      Object {
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#EEE",
          "fontFamily": "Gilroy-Regular",
          "fontSize": 20,
          "lineHeight": 32,
          "textAlign": undefined,
          "textTransform": undefined,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(51, 51, 51, 0.8)",
        "justifyContent": "center",
        "opacity": 1,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#EEE",
          "fontFamily": "Gilroy-Regular",
          "fontSize": 24,
          "lineHeight": 32,
          "textAlign": undefined,
          "textTransform": undefined,
        }
      }
    >
      Try again
    </Text>
  </View>
</View>
`;
